core/Plugin/Metric.php

```diff
@@ -108,26 +108,41 @@ public function beforeFormat($report, DataTable $table)
      */
     public static function getMetric($row, $columnName, $mappingNameToId = null)
     {
-        if (empty($mappingNameToId)) {
-            $mappingNameToId = Metrics::getMappingFromNameToId();
-        }
-
         if ($row instanceof Row) {
             $value = $row->getColumn($columnName);
-            if ($value === false
-                && isset($mappingNameToId[$columnName])
-            ) {
-                $value = $row->getColumn($mappingNameToId[$columnName]);
+
+            if ($value === false) {
```

Comment:
It does actually not often happen that we do not have a value therefore get the MappingFromNameToId only if needed => performance win